content/v2.0/router/advanced-usage.md

```diff
@@ -0,0 +1,624 @@
+---
+title: "Router: Advanced Usage"
+order: 30
+---
+
+
+[Refer](https://github.com/hanami/router/blob/unstable/spec/integration/hanami/router/mount_spec.rb)
+
+## Mounting an application
+
+In Hanami, you can mount any rack application in desired route. If we want to do it, you should use the `mount` keyword.
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+  mount Api::App.new, at: "/api"
+  mount Backend::App, at: "/backend"
+  mount \->(\*) { \[200, {"Content-Length" => "4"}, \["proc"\]\] }, at: "/proc"
+  mount \->(\*) { \[200, {"Content-Length" => "8"}, \["trailing"\]\] }, at: "/trailing/"
+end
+```
+
+#### Mounting To A Path
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+  mount SinatraApp.new, at: "/sinatra"
+end
+```
+
+All the HTTP requests starting with `/sinatra` will be routed to `SinatraApp`.
+
+#### Mounting On A Subdomain
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+    mount Blog.new, host: "blog"
+end
+```
+
+All the HTTP requests to `http://blog.example.com` will be routed to `Blog`.
+
+<!--   TODO: Update this notice to meet 2.0 behaviour
+<p class="notice">
+  In development, you will NOT be able to access <code>http://blog.localhost:2300</code>,
+  so you should specify a host when running the server:
+  <code>bundle exec hanami server --host=lvh.me</code>.
+  Then your application can be visited at <code>http://blog.lvh.me:2300</code>
+</p>
+-->
+
+### RESTful Resources
+
+Hanami has native [REST](http://en.wikipedia.org/wiki/Representational_state_transfer) support.
+
+At the routing level, there are two methods that can be used to declare them: `resources` and `resource`.
+The former is for plural resources, the latter for singular ones.
+
+Declaring a resource means to generate **several default routes** with just one line of code.
+
+## RESTful Resources
+
+### Default Routes
+
+```ruby
+# /config/routes.rb
+
+Hanami.applicaton.routes do
+  resources :books
+end
+
+```
+
+It generates:
+
+<table class="table table-bordered table-striped">
```

Comment:
Can we change this to (GitHub-flavored) Markdown tables? https://github.github.com/gfm/#tables-extension-

Reply:
Sure. I copied it from existing docs, but if other syntax works better, I don't mind.